File: packages/shared/src/inbox.ts
// Sprint 2: shared inbox types. Mirror the documents_inbox table and the
// Pydantic models in apps/api/app/models/inbox.py.

export type OcrStatus = "pending" | "in_progress" | "completed" | "failed";

export type ParseStatus =
  | "pending"
  | "in_progress"
  | "completed"
  | "failed"
  | "needs_review";

export type RoutingStatus =
  | "pending"
  | "routed"
  | "client_not_found"
  | "new_gst_registration_detected"
  | "pan_gstin_mismatch"
  | "no_identifier_found"
  | "manual_assignment";

export type IngestChannel =
  | "web_upload"
  | "mobile_capture"
  | "email"
  | "gst_portal_gsp"
  | "it_portal_aa"
  | "whatsapp";

export interface InboxItem {
  inbox_id: string;
  original_filename: string;
  file_size_bytes: number;
  page_count: number | null;
  mime_type: string | null;
  ocr_status: OcrStatus;
  ocr_provider_used: string | null;
  ocr_error: string | null;
  ingest_channel: IngestChannel;
  parse_status: ParseStatus;
  routing_status: RoutingStatus;
  uploaded_at: string;
}

export interface InboxList {
  items: InboxItem[];
  total: number;
}

export interface UploadResponse {
  inbox_id: string;
  ocr_status: OcrStatus;
  file_hash: string;
  file_size_bytes: number;
}

export interface OcrTextResponse {
  inbox_id: string;
  ocr_status: OcrStatus;
  ocr_text: string | null;
  ocr_provider_used: string | null;
  page_count: number | null;
  ocr_error: string | null;
}

export const INGEST_CHANNEL_LABELS: Record<IngestChannel, string> = {
  web_upload: "Upload",
  mobile_capture: "Mobile",
  email: "Email",
  gst_portal_gsp: "GST portal",
  it_portal_aa: "IT portal",
  whatsapp: "WhatsApp",
};

export const OCR_STATUS_LABELS: Record<OcrStatus, string> = {
  pending: "Uploading",
  in_progress: "OCR in progress",
  completed: "Ready for review",
  failed: "OCR failed",
};

File: apps/api/app/models/inbox.py
"""Pydantic models for the documents_inbox API surface."""

from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field

OcrStatus = Literal["pending", "in_progress", "completed", "failed"]
ParseStatus = Literal["pending", "in_progress", "completed", "failed", "needs_review"]
RoutingStatus = Literal[
    "pending",
    "routed",
    "client_not_found",
    "new_gst_registration_detected",
    "pan_gstin_mismatch",
    "no_identifier_found",
    "manual_assignment",
]
IngestChannel = Literal[
    "web_upload",
    "mobile_capture",
    "email",
    "gst_portal_gsp",
    "it_portal_aa",
    "whatsapp",
]


class InboxItem(BaseModel):
    inbox_id: UUID
    original_filename: str
    file_size_bytes: int
    page_count: int | None = None
    mime_type: str | None = None
    ocr_status: OcrStatus
    ocr_provider_used: str | None = None
    ocr_error: str | None = None
    ingest_channel: IngestChannel
    parse_status: ParseStatus
    routing_status: RoutingStatus
    routing_anomaly_details: dict[str, object] | None = None
    parsed_to_notice_id: UUID | None = None
    matched_client_name: str | None = None
    matched_registration_label: str | None = None
    document_type: str | None = None
    parse_confidence: float | None = None
    uploaded_at: datetime


class InboxList(BaseModel):
    items: list[InboxItem]
    total: int


class UploadResponse(BaseModel):
    inbox_id: UUID
    ocr_status: OcrStatus = "pending"
    file_hash: str
    file_size_bytes: int


class OcrTextResponse(BaseModel):
    inbox_id: UUID
    ocr_status: OcrStatus
    ocr_text: str | None
    ocr_provider_used: str | None
    page_count: int | None
    ocr_error: str | None


class EmailAttachment(BaseModel):
    filename: str
    mime_type: str
    content_b64: str = Field(..., description="Base64-encoded attachment bytes")


class EmailInboundPayload(BaseModel):
    sender: str
    recipient: str
    subject: str | None = None
    received_at: datetime | None = None
    attachments: list[EmailAttachment] = Field(default_factory=list)


class EmailInboundResponse(BaseModel):
    accepted: int
    skipped: int
    inbox_ids: list[UUID]
